### ISSUE TITLE
Vim install gets blocked behind a GUI error dialog

### ISSUE THREAD

**Issue by jazzdelightsme** [microsoft/winget-pkgs#19866]
# Brief description of your issue
I tried installing vim, but the setup process was blocked on a GUI error dialog. Don't know if this is a problem with the winget manifest or the vim installer or what.

# Steps to reproduce

1. Non-elevated powershell prompt.
2. `winget install --id vim.vim --silent --location C:\vim`
3. Click "yes" on the elevation prompt.

# Expected behavior
Vim install process should complete without problems.

# Actual behavior
Shortly after starting, the install process is blocked by a GUI popup:

```
[Window Title]
C:\vim\vim82\uninstall-gui.exe

[Content]
Windows cannot find 'C:\vim\vim82\uninstall-gui.exe'. Make sure you typed the name correctly, and then try again.

[OK]
```

# Environment

I see that the vim package being installed is version 8.2.3107. And I think it was recently updated by @ImJoakim .

```none
Windows Package Manager v1.0.11692
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22410.1000
Package: Microsoft.DesktopAppInstaller v1.12.11692.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir
```

**Comment by jedieaston** [comment]
I cannot replicate this exact error in a Sandbox:
```
Windows Package Manager v1.0.11692
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22000.51
Package: Microsoft.DesktopAppInstaller v1.12.11692.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

Links
--------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget

```

However, vim does not seem to use the location in the `--location` argument on my system. What determines if that is supported, does anyone know?

**Comment by OfficialEsco** [comment]
> However, vim does not seem to use the location in the `--location` argument on my system. What determines if that is supported, does anyone know (is it an installer type or what?)

I have not experimented with this, but i guess it requires the InstallLocation switch to be configured?
https://github.com/microsoft/winget-pkgs/blob/master/doc/manifest/schema/1.0.0/installer.md#installlocation
https://raw.githubusercontent.com/microsoft/winget-cli/master/schemas/JSON/manifests/v1.0.0/manifest.installer.1.0.0.json

**Comment by jedieaston** [comment]
Oh, so this is why people added the `<INSTALLPATH>` arg to Visual Studio. I thought that would cause issues.

**Comment by jazzdelightsme** [comment]
I discovered that the system that I ran this on had previously had vim installed, but the directory (C:\vim) had been wiped out. On a fresh VM, I found that I could install vim, and was able to repro the problem by manually deleting the vim directory and re-running the install. So I think the problem from the OP is a problem with the vim installer itself (or perhaps Nullsoft or whatever stack it uses), *not* the winget package manifest.

I will open a separate Issue for `--location` support.

Thanks!
